Migration article VPC peering - General

dewan-ahmed opened the pull request:
# What changed, and why it matters

Resolves https://github.com/aiven/devportal/issues/796 and all three sub-tasks

This PR brings [Using VPC Peering](https://help.aiven.io/en/articles/778836-using-virtual-private-cloud-vpc-peering) and [Azure VPC Peering](https://help.aiven.io/en/articles/3435096-azure-virtual-network-peering) articles to DevPortal.

Important notes to the reviewer:

1. The reader will be able to follow the instructions without the screenshots. Based on the contribution policy around screenshots, I didn't include any. If you feel that one or more screenshot is required for ease of understanding, please let me know.
2. I have validated the steps for GCP VPC peering. I suggest to have one or more SA validate the VPC peering for AWS and Azure. It's faster if someone has the access and setup already.
3. Since the changes are related and cross-referenced, I didn't break down into smaller PRs. The VPC and cloud-specific peering articles should go live at the same time.
4. Please update the relevant migration spreadsheets after merging this PR. If unsure, please let me know and I'll take care of that step.

dewan-ahmed commented:
I'm working on fixing the Vale warnings now. Another commit will follow in a few hours. Marking the PR **draft** till then.

dewan-ahmed commented:
Hi @laysauchoa adding you as a reviewer per your request. You can either review this PR partially for what Issue #1106 covers OR you can review the whole PR :)

Do you have access to an Azure account? Then you could validate the Azure part of the article as well.

This is still in draft so you'll get a notification once the PR is ready to be reviewed. Thank you.

laysauchoa commented:
Hi Dewan, thank you for your nice contribution. Would be possible to split this in three PRs? I think it makes easier for others to review and they can be reviewed independent. I'll only be able to review the AWS article due to time, so please, feel free to add others that can assist you with this review.

dewan-ahmed commented:
Thanks @laysauchoa for kindly offering to review the AWS part of the PR. Due to the cross-links among these PRs and the time crunch, I propose to use this single PR for these VPC-related articles (this is an exception). 

I tested the GCP part of the article. If you test the AWS part, I'll request help from SA colleagues to have the Azure part reviewed. Thanks again.

dewan-ahmed commented:
Hi @vietnguyen-aiven I added a note via https://github.com/aiven/devportal/pull/1310/commits/51b7f7234b982d9cab78152b347d431cc11e965d based on your comment on https://github.com/aiven/devportal/issues/796.

ftisiot commented:
Should we ask the SA team to validate the steps to keep this moving? I personally don't have the access to all the clouds to do it.

dewan-ahmed commented:
I recently got access to Azure so I can test the Azure steps today. I've already verified the GCP steps. I believe @laysauchoa has AWS access. Which cloud do you have access to @ftisiot?

dewan-ahmed commented:
I'm currently blocked with the Azure testing part of this article. Here are the details:

1. `docs/platform/howto/setup-vpc-peering-azure` seems very tedious compared to VPC peering steps in other clouds. Are we sure that there are no other easier/shorter ways? Here is my understanding/**expectation** of the flow:  
---There is an Aiven network on the given cloud. We go to **VPC --> VPC Peering connections** from the Aiven console and add the Azure Subscription ID, Resource group, Network name, Active Directory tenant ID, and Application Object ID to request the peering connection. The first four parameters should already be known and we can create an Azure Application to get the Object ID. Once the peering connection is request, one can approve from the Azure console or CLI for the peering connection to be successful.--- Is the actual process different which requires all the steps mentioned in the help article?
2. For `az ad sp create --id 55f300d4-fc50-4c5e-9222-e90a6e2187fb` step, is this static ID? Using this command is getting me an error:
```
When using this permission, the backing application of the service principal being created must in the local tenant
```
3. For `az ad app create --display-name "<name of your choosing>" --available-to-other-tenants --key-type Password` command, `--available-to-other-tenants` flag is not a valid flag per https://docs.microsoft.com/en-us/cli/azure/ad/app?view=azure-cli-latest#az-ad-app-create

I'm looking for some help validating the Azure VPC peering part of this article.

dewan-ahmed commented:
Hi @laysauchoa @ftisiot since I'm blocked on the Azure part of this PR, I've divided this PR into four parts (Laysa already suggested this 🙂 ). You can review the current PR for general VPC creation and I'll raise subsequent PRs for AWS and GCP. Once I get some help with Azure, I'll raise Azure VPC peering PR. Thanks.

laysauchoa commented:
Dewan, I reviewed the article and test it out. All looks good!!!

About the article structure, the title is about "Create VPC" but there is more content on this article than that like "Deploy new services to a VPC", "Migrate a public service to a VPC", "Accessing VPC services from the public internet", "Security"...

I wonder if one of the following would improve the article structure: HowTo > "Manage Virtual Private Cloud (VPC) peering":
- "Create VPC peering"
- "Deploy new services to a VPC"
- "Migrate a public service to a VPC"
- "Accessing VPC services from the public internet"
- "Security" maybe can be in the introduction or in some subsection?

When I think about creation of services, I think about deletion. So I was wondering if it would make sense to add some information on that. I see that you can click and delete but what does it happen with the services that are using the VPC after the deletion? 

Let me know what you think or what are your ideas about it.

dewan-ahmed commented:
Thanks @laysauchoa for the insightful feedback. I've modified the article title and added a deletion section per your suggestion. I've also removed the security section and added the information within another section.

ftisiot commented on docs/platform/howto/manage-vpc-peering.rst:
```diff
@@ -0,0 +1,78 @@
+.. _platform_howto_setup_vpc_peering:
+
+Manage Virtual Private Cloud (VPC) peering
```
there's a warning here about title not being in sentence case

ftisiot reviewed (changes requested):
Small edit :D

ftisiot commented:
Should we ask the SAs to validate? I miss the related knowledge

dewan-ahmed commented:
Hi @ftisiot I've verified the steps for this article which is only for the Aiven platform. There will be three follow-up PRs to include cloud-specific VPC peering and we can request an SA to validate those ones.

laysauchoa reviewed (approved):
This is good and I followed the steps and worked for me. Thanks a lot @dewan-ahmed for bringing this article to live in the docs aiven io.
